pkg/tfgen/generate_nodejs.go

```diff
@@ -475,7 +475,17 @@ func (g *nodeJSGenerator) emitPlainOldType(w *tools.GenWriter, pot *plainOldType
 func (g *nodeJSGenerator) emitResourceType(mod *module, res *resourceType) (string, error) {
 	// Create a resource module file into which all of this resource's types will go.
 	name := res.name
-	w, file, err := g.openWriter(mod, lowerFirst(name)+".ts", true)
+
+	filename := name
+
+	// We need to check if the resource is called Index. If so then we will have problems
+	// based on the fact that we need to generate an Index.ts based on the package contents
+	// Therefore, we are going to append _ into the name to allow us to continue
+	if filename == "Index" {
+		filename = fmt.Sprintf("%s_", name)
+	}
```

Comment:
It’s logically the same, but seems cleaner to do the lowerFirst first, then compare with the “reserved filenames (`index.ts`, I think there are others?)?

Reply:
Yes, we should also look for `utilities.ts`

Reply:
Fixed and pushed - also checking for utilities as that is also a reserved filename